src/System.Private.CoreLib/shared/System/Numerics/Vector.cs

```diff
@@ -778,7 +778,65 @@ private Vector(ref Register existingRegister)
 
 #if netcoreapp
         /// <summary>
-        /// Constructs a vector from the given span. The span must contain at least Vector'T.Count elements.
+        /// Constructs a vector from the given <see cref="ReadOnlySpan{Byte}"/>. The span must contain at least <see cref="Vector{Byte}.Count"/> elements.
+        /// </summary>
+        public Vector(ReadOnlySpan<byte> values)
+            : this()
+        {
+            if ((typeof(T) == typeof(byte))
+                || (typeof(T) == typeof(sbyte))
+                || (typeof(T) == typeof(ushort))
+                || (typeof(T) == typeof(short))
+                || (typeof(T) == typeof(uint))
+                || (typeof(T) == typeof(int))
+                || (typeof(T) == typeof(ulong))
+                || (typeof(T) == typeof(long))
+                || (typeof(T) == typeof(float))
+                || (typeof(T) == typeof(double)))
+            {
+                if (values.Length < Vector<byte>.Count)
+                {
+                    throw new IndexOutOfRangeException(SR.Format(SR.Arg_InsufficientNumberOfElements, Vector<byte>.Count, nameof(values)));
```

Comment:
:usual_suspects: Put throw statements to  static methods. It will make body of the method much smaller.